File: src/main/scala-graphql/io/github/graphql/scala/model/ContributionOrderTO.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.Objects
import scala.collection.JavaConverters._
import OrderDirectionTO._
import ContributionOrderFieldTO._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:37+0800"
)
class ContributionOrderTO(
    @javax.validation.constraints.NotNull
    direction: OrderDirectionTO,
    field: ContributionOrderFieldTO
) {

    override def toString(): String = {
        Seq(
            if (direction != null) "direction: " + GraphQLRequestSerializer.getEntry(direction) else "",
            if (field != null) "field: " + GraphQLRequestSerializer.getEntry(field) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }

    override def equals(obj: Any): Boolean = {
        if (this == obj) {
            return true
        }
        if (obj == null || getClass != obj.getClass) {
            return false
        }
        val that = obj.asInstanceOf[ContributionOrderTO]
        Objects.equals(direction, that.direction) &&
        Objects.equals(field, that.field)
    }

    override def hashCode(): Int = {
        Objects.hash(direction, field)
    }
}

object ContributionOrderTO {

    def builder(): ContributionOrderTO.Builder = new Builder()

    class Builder {

        private var direction: OrderDirectionTO = _
        private var field: ContributionOrderFieldTO = _

        def setDirection(direction: OrderDirectionTO): Builder = {
            this.direction = direction
            this
        }

        def setField(field: ContributionOrderFieldTO): Builder = {
            this.field = field
            this
        }

        def build(): ContributionOrderTO = new ContributionOrderTO(direction, field)

    }
}

File: src/main/kotlin-graphql/io/github/graphql/model/ContributionOrderTO.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:04+0800"
)
open class ContributionOrderTO(
    val direction: OrderDirectionTO,
    val field: ContributionOrderFieldTO?
) {

    companion object {
        fun builder(): Builder = Builder()
    }

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        joiner.add("direction: " + GraphQLRequestSerializer.getEntry(direction))
        if (field != null) {
            joiner.add("field: " + GraphQLRequestSerializer.getEntry(field))
        }
        return joiner.toString()
    }

    class Builder {

        private lateinit var direction: OrderDirectionTO
        private var field: ContributionOrderFieldTO? = null

        fun setDirection(direction: OrderDirectionTO): Builder {
            this.direction = direction
            return this
        }

        fun setField(field: ContributionOrderFieldTO?): Builder {
            this.field = field
            return this
        }

        fun build(): ContributionOrderTO {
            return ContributionOrderTO(direction, field)
        }
    }
}
